src/nvim/ex_cmds.h

```diff
@@ -19,13 +21,47 @@
 #define ECMD_LAST       (linenr_T)-1    /* use last position in all files */
 #define ECMD_ONE        (linenr_T)1     /* use first line */
 
+/// for cmdl_progress in incsubstitute
+typedef enum {
+  ICS_NO_WD,                 /// state of cmdline when none words are typed
+                              /// ":%s" or ":%s/"
+  ICS_ONE_WD,                /// state of cmd line when only pattern word began
+                              /// to be typed : ":%s/patt"
+  ICS_TWO_SLASH_ONE_WD,      /// Second / has been typed but not the second
+                              /// word yet : "%s/pattern/"
+  ICS_TWO_WD                /// state of cmd line when pattern was completed
+                            /// and substitue is being typed : ":%s/pattern/sub"
+} IncSubstitute_state;
+
 /// Previous :substitute replacement string definition
 typedef struct {
   char *sub;            ///< Previous replacement string.
   Timestamp timestamp;  ///< Time when it was last set.
   list_T *additional_elements;  ///< Additional data left from ShaDa file.
 } SubReplacementString;
 
+/// Defs for inc_sub functionality
```

Comment:
This documents *just* `_noop` as “defs for inc_sub functionality”. Obviously wrong, use `//` and better put an empty line after the comment.